src/main/jbake/content/developer/jms/index.adoc

```diff
@@ -0,0 +1,64 @@
+
+
+=== Glossary
+
+
+* *JMS Provider*: The implementation of Message Oriented Middleware following JMS interfaces.
+* *JMS Client*:  Application that produces or/and receives messages.
+* *JMS Producer*: JMS client that creates and sends messages.
+* *JMS Consumer*: JMS client that receives messages.
+* *JMS Message*: The object that is transferred between clients.
+* *JMS Queue*: Area that contains sent messages that are waiting to be read by only one consumer.
+* *JMS Topic*: A distribution mechanism for publishing messages that are delivered to multiple subscribers.
+
+
+
+=== Message
+
+The Message interface is the root interface of all JMS messages. It defines the message header and the acknowledge method used for all messages.
+
+*More*: https://docs.oracle.com/javaee/7/api/javax/jms/Message.html
+
+==== Message Compose
+
+* *Header*: All messages support the same set of header fields. Header fields contain values used by both clients and providers to identify and route messages.
+* *Properties*: Each message contains a built-in facility for supporting application-defined property values. Properties provide an efficient mechanism for supporting application-defined message filtering.
+* *Body*: The JMS API defines several types of message body, which cover the majority of messaging styles currently in use.
+
+==== Message Bodies
+
+* *Stream*: A *StreamMessage* object's message body contains a stream of primitive values in the Java programming language ("Java primitives"). It is filled and read sequentially.
+* *Map*: A *MapMessage* object's message body contains a set of name-value pairs, where names are String objects, and values are Java primitives. The entries can be accessed sequentially or randomly by name. The order of the entries is undefined.
+* *Text*: A *TextMessage* object's message body contains a java.lang.String object. This message type can be used to transport plain-text messages, and XML messages.
+* *Object*: An *ObjectMessage* object's message body contains a Serializable Java object.
+* *Bytes*: A *BytesMessage* object's message body contains a stream of uninterpreted bytes. This message type is for literally encoding a body to match an existing message format. In many cases, it is possible to use one of the other body types, which are easier to use. Although the JMS API allows the use of message properties with byte messages, they are typically not used, since the inclusion of properties may affect the format.
+
+
+=== JMS and TomEE
+
+Apache TomEE (Plume and Plus) provide Apache ActiveMQ (http://activemq.apache.org/) as part of the distribution,
+ and this can run in either embedded mode, or TomEE can connect to an external standalone ActiveMQ broker.
+Apache TomEE does allow ActiveMQ to be replaced by another JMS provider, or alternatively you can run two or more
+ JMS providers simultaneously and specify which provider MDBs and callers should use via configuration.
+
```

Comment:
I'd suggest including, or linking to, the basic configs for ActiveMQ in TomEE here. Essentially, you want the information that is here: http://tomee.apache.org/jms-resources-and-mdb-container.html